openstack/nova/templates/api-deployment.yaml

```diff
@@ -90,58 +90,10 @@ spec:
             - name: nova-api
               containerPort: {{.Values.global.novaApiPortInternal}}
           volumeMounts:
-            - mountPath: /etc/nova
-              name: etcnova
-            - mountPath: /etc/nova/nova.conf
-              name: nova-etc
-              subPath: nova.conf
-              readOnly: true
-            - mountPath: /etc/nova/nova-api.conf
-              name: nova-etc
-              subPath: nova-api.conf
-              readOnly: true
-            - mountPath: /etc/nova/api-paste.ini
-              name: nova-etc
-              subPath: api-paste.ini
-              readOnly: true
-           {{- if .Values.api.use_uwsgi }}
-            - mountPath: /etc/nova/api_uwsgi.ini
-              name: nova-etc
-              subPath: api_uwsgi.ini
-              readOnly: true
-           {{- end }}
-{{- if .Values.audit.enabled }}
-            - mountPath: /etc/nova/nova_audit_map.yaml
-              name: nova-etc
-              subPath: nova_audit_map.yaml
-              readOnly: true
-{{- end }}
-            - mountPath: /etc/nova/policy.json
-              name: nova-etc
-              subPath: policy.json
-              readOnly: true
-            - mountPath: /etc/nova/logging.ini
-              name: nova-etc
-              subPath: logging.ini
-              readOnly: true
-            - mountPath: /etc/nova/rootwrap.conf
-              name: nova-etc
-              subPath: rootwrap.conf
-              readOnly: true
-            - mountPath: /etc/nova/rootwrap.d/api-metadata.filters
-              name: nova-etc
-              subPath: api-metadata.filters
-              readOnly: true
-            - mountPath: /var/lib/kolla/venv/bin/iptables-restore
-              name: nova-bin
-              subPath: iptables-restore.mock
-              readOnly: true
-            {{- if .Values.watcher.enabled }}
             - name: nova-etc
-              mountPath: /etc/nova/watcher.yaml
-              subPath: watcher.yaml
-              readOnly: true
-            {{- end }}
+              mountPath: /etc/nova
+            - name: nova-bin
+              mountPath: /var/lib/kolla/venv/bin
```

Comment:
Did you confirm this works? Is everything else still there in `/var/lib/kolla/venv/bin`?

Reply:
It seems that here we need to keep using `subPath` otherwise the `bin` directory will get overwritten. I think there's no problem with that, since this is the only place where nova-bin is being used. Then does it still make sense to project a volume here? Probably not.